### PR TITLE
Adding support for https connections

### PR DESCRIPTION
**Pull request by No-Signal**
Adding support for https connections and ability to ignore unauthorized requests e.g. ssl certificate issues for self signed certificates

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,8 +38,10 @@ Submitting or debugging ECL using VS-Code requires specificying the target envir
 	"mode": "submit",
 	"workspace": "${workspaceRoot}",
 	"program": "${file}",
+    "protocol": "http",
```
Comment (GordonSmith) [inline README.md]: Minor - tab v spaces

File: package.json [hunk 2]
```diff
@@ -124,7 +124,9 @@
                         "required": [
                             "workspace",
                             "serverAddress",
+                            "protocol",
                             "port",
+                            "rejectUnauthorized",
```
Comment (GordonSmith) [inline package.json]: Minor:  Is this really required, a default of false would probably be easier for the user?

File: package.json [hunk 3]
```diff
@@ -150,11 +152,21 @@
                                 "description": "Server IP/Domain address.",
                                 "default": "127.0.0.1"
                             },
+                            "protocol": {
+                                "type": "string",
+                                "description": "Server protocol (http or https).",
```
Comment (GordonSmith) [inline package.json]: MInor:  Could have an "enum":
```
"enum": [
    "http",
    "https"
]
```

File: package.json [hunk 4]
```diff
@@ -150,11 +152,21 @@
...
+                            },
                             "port": {
                                 "type": "number",
                                 "description": "The WsWorkunits port.",
                                 "default": 8010
                             },
+                            "rejectUnauthorized": {
+                                "type": "boolean",
+                                "description": "Reject unauthorized calls e.g. SSL certificate errors.",
+                                "default": true
```
Comment (GordonSmith) [inline package.json]: Minor:  Default false would be safer?

File: package.json [hunk 5]
```diff
@@ -204,8 +216,10 @@
                         "mode": "submit",
                         "workspace": "${workspaceRoot}",
                         "program": "${file}",
+                        "protocol": "http",
                         "serverAddress": "localhost",
                         "port": 8010,
+                        "rejectUnauthorized": true,
```
Comment (GordonSmith) [inline package.json]: Minor: Default false would be safer?

File: package.json [hunk 6]
```diff
@@ -226,8 +240,10 @@
                             "mode": "submit",
                             "workspace": "^\"\\${workspaceRoot}\"",
                             "program": "${file}",
+                            "protocol": "http",
                             "serverAddress": "localhost",
                             "port": 8010,
+                            "rejectUnauthorized": true,
```
Comment (GordonSmith) [inline package.json]: Minor: Default false would be safer?

**Comment by GordonSmith** [comment]
@No-Signal Thanks for the PR (you forgot to post an obligatory "first"!).  Few minor comments which I will fix prior to releasing.

FYI I am in the process of ripping out the "esp comms" layer from:
* VS-Code
* ECL Watch
* Visualizations
...and combining them into a new package (name subject to change) hpcc-platform-comms:  https://github.com/GordonSmith/hpcc-platform-comms which will replace some of the above.

File: package.json (hunk 4 above)
Comment (No-Signal) [inline package.json]: The default value for rejectUnauthorized in the request object is true so i just maintained the default behaviour

Comment (GordonSmith) [inline package.json]: Good to know - I am pretty sure I can't use the node "request" lib within a browser environment - so its going to change anyway.

**Pull request merged**